halo2_proofs/benches/poly.rs

```diff
@@ -0,0 +1,39 @@
+#[macro_use]
+extern crate criterion;
+
+use crate::arithmetic::{compute_inner_product, eval_polynomial};
+use crate::pasta::Fp;
+use group::ff::Field;
+use halo2_proofs::*;
+
+use criterion::{BenchmarkId, Criterion};
+use rand_core::OsRng;
+
+fn criterion_benchmark(c: &mut Criterion) {
+    let mut eval_polynomial_group = c.benchmark_group("poly-eval_polynomial");
+    for k in 3..19 {
+        eval_polynomial_group.bench_function(BenchmarkId::new("k", k), |b| {
+            b.iter(|| {
+                let poly = (0..(1 << k)).map(|_| Fp::random(OsRng)).collect::<Vec<_>>();
+                let point = Fp::random(OsRng);
+                eval_polynomial(&poly, point);
+            });
```

Comment:
This is currently including the cost of randomness generation and allocations in the benchmark. Move `poly` and `point` out of the hot loop closure:
```suggestion
            let poly = (0..(1 << k)).map(|_| Fp::random(OsRng)).collect::<Vec<_>>();
            let point = Fp::random(OsRng);
            b.iter(|| {
                eval_polynomial(&poly, point);
            });
```